src/components/CryptoIcon/CryptoIcon.js

```diff
@@ -1,168 +1,83 @@
-    REN: renIcon,
-    RAI: raiIcon,
-    ENS: ensIcon,
-    UST: ustIcon,
-    rETH: rethIcon,
-    cbETH: cbethIcon,
-    stkAAVE: stkAaveIcon,
-    WAVAX: avaxIcon,
-    sAVAX: savaxIcon,
-    CVX: cvxIcon,
-    MAI: maiIcon,
-    "1INCH": oneinchIcon,
-    LUSD: lusdIcon,
-    MATIC: maticIcon,
-    polygon: maticIcon,
-    ethereum: ethIcon,
-    AVAX: avaxIcon,
-    arbitrum: arbitrumIcon,
-    avalanche: avaxIcon,
-    optimism: optimismIcon,
-    LDO: ldoIcon,
-    GHST: ghstIcon,
-    MaticX: maticxIcon,
-    stMATIC: stmaticIcon,
-    WMATIC: wmaticIcon,
-    miMATIC: maiIcon,
-    jEUR: jeurIcon,
-    agEUR: ageurIcon,
-  };
-  const icon = mapping[name];
-  if (!icon) {
+  const { name, size, text, ...rest } = props;
+
+  const uniqueIdentifier = `icon-${_.uniqueId()}`;
+
+  const [isLoaded, setIsLoaded] = useState(false);
+
+  let updatedName = name;
+
+  if (name === "ethereum") {
+    updatedName = "eth";
+  } else if (name === "avalanche") {
+    updatedName = "avax";
+  } else if (name === "polygon") {
+    updatedName = "matic";
+  } else if (name === "arbitrum") {
+    updatedName = "arb";
+  } else if (name === "optimism") {
+    updatedName = "op";
+  } else if (name === "xSUSHI") {
+    updatedName = "sushi";
+  } else if (name === "WETH.e") {
+    updatedName = "weth";
+  } else if (name === "WBTC.e") {
+    updatedName = "wbtc";
+  } else if (name === "USDC.e") {
+    updatedName = "usdc";
+  } else if (name === "USDT.e") {
+    updatedName = "usdt";
+  } else if (name === "DAI.e") {
+    updatedName = "dai";
+  } else if (name === "WAVAX") {
+    updatedName = "avax";
+  } else if (name === "AAVE.e") {
+    updatedName = "aave";
+  } else if (name === "LINK.e") {
+    updatedName = "link";
+  } else if (name === "BTC.b") {
+    updatedName = "btc";
+  } else if (name === "sAVAX") {
+    updatedName = "avax";
+  } else if (name === undefined) {
     return null;
+  } else if (name === "stMATIC") {
+    updatedName = "matic";
   }
-  let component;
-  if (address) {
-    const etherscanUrl = smartEtherscanUrl(location);
-    const link = `${etherscanUrl}address/${address}`;
-    component = (
-      <a href={link}>
-        <img {...rest} src={icon} alt={name} style={{ width: size, height: size }} />
-      </a>
-    );
-  } else {
-    component = (
-      <img {...rest} src={icon} alt={name} style={{ width: size, height: size }} />
-    );
-  }
-  return component;
+
+  const id = `icon-${updatedName.toLowerCase()}-${uniqueIdentifier}`;
+
+  const baseUrl = "https://icons.blockanalitica.com/currency/";
+  const src = `${baseUrl}${updatedName.toLowerCase()}.svg`;
+
+  return name ? (
+    <>
+      <img
+        {...rest}
+        id={id}
+        src={src}
+        alt={updatedName}
+        style={{ width: size, height: size }}
+        onLoad={() => setIsLoaded(true)}
+      />
+      {isLoaded && (
+        <UncontrolledTooltip placement="bottom" target={id}>
+          {text ? `${text} ${updatedName}` : updatedName}
```

Comment:
This tooltip should show the unmodified `name` instead of `updatedName` so we'd actually show the name of the token, not the filename of the token